ipa-core/src/protocol/ipa_prf/prf_sharding/bucket.rs

```diff
@@ -76,17 +81,16 @@ impl From<MoveToBucketError> for Error {
 ///
 /// ## Errors
 /// If `breakdown_count` does not fit into `BK` bits or greater than or equal to $2^9$
-pub async fn move_single_value_to_bucket<C, S, F>(
-    ctx: C,
+pub async fn move_single_value_to_bucket<'a, F>(
+    ctx: UpgradedSemiHonestContext<'a, NotSharded, F>,
```

Comment:
I wrote `NotSharded` because it was the easy thing to do, but we'll probably need to change this to `B: ShardBinding` fairly soon. But I don't think it's important to update in this PR.

(I left a comment on one of the sharding PRs about the safety risk of protocols having access to the shard API -- it occurs to me that writing against `B: ShardBinding` is actually reasonably safe, in that if your protocol is generic over sharded or not, then you probably can't access the sharding API.)